codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieDecoder.java

```diff
@@ -32,47 +32,61 @@
     private final InternalLogger logger = InternalLoggerFactory.getInstance(getClass());
 
     private final boolean strict;
+    private final boolean reportException;
+    private final boolean needReport;
+
+    protected CookieDecoder(boolean strict, boolean reportException) {
+       this.strict = strict;
+       this.reportException = reportException;
+       needReport = reportException || logger.isDebugEnabled();
+    }
 
     protected CookieDecoder(boolean strict) {
-        this.strict = strict;
+        this(strict, false);
     }
 
     protected DefaultCookie initCookie(String header, int nameBegin, int nameEnd, int valueBegin, int valueEnd) {
         if (nameBegin == -1 || nameBegin == nameEnd) {
-            logger.debug("Skipping cookie with null name");
+            if (needReport) {
+                reportError(String.format("Invalid cookie with null name found in header '%s'", header));
```

Comment:
There is another option with override..

in the `CookieDecoder`:
```java
protected boolean isNeedReportError() {
    return logger.isDebugEnabled();
}

protected void reportError(String message) {
    logger.debug(message);
}

...
if (isNeedReportError()) {
    reportError(String.format("..."));
}
...
```
in the `ServerCookieDecoder`:
```java
public static final ServerCookieDecoder FAIL_FAST = new ServerCookieDecoder(true) {
    @Override
    protected boolean isNeedReportError() {
        return true;
    }

    @Override
    protected void reportError(String message) {
        throw new IllegalArgumentException(message);
    }
}
```


I like this more.